### DISCUSSION TITLE
shallowRef in Pinia store accessible via .value property?

### DISCUSSION THREAD

**Discussion by robertgirschick** [vuejs/pinia#1614] [Help and Questions]
I have some `shallowRef` state variables which I need to pass along in functions, so `shallowReactive` is not an option. When I use local variables, a receiving function can access the `value` property and mutate it easily:

```
const localState = shallowRef({
  one: {
    two: {
      three: "test"
    },
  },
});
```

Then in function: `passedAlongState.value = {...}` works fine, it is accessible and still reactive.

If I put the `shallowRef` above into a Pinia store, the function doesn't work, because it can't access the state via `.value`. Is there a way to get `shallowRef`s in a Pinia store back just like "normal" local state, that is accessible via `.value` while keeping reactivity?

**Comment by Matthew-E-Gould** [comment]
what processor do you have on your PC? have the same issue that only affected AMD-based machines.

**Comment by posva** [comment]
the refs gets unwrapped so it should keep the reactivity. You will have to share more code in order to be able to tell 👀
